extensions/vscode-api-tests/src/extension.ts

```diff
@@ -796,16 +796,52 @@ export class MemFS implements vscode.FileSystemProvider, vscode.FileSearchProvid
 	// --- search provider
 
 	provideFileSearchResults(query: vscode.FileSearchQuery, _options: vscode.FileSearchOptions, _token: vscode.CancellationToken): vscode.ProviderResult<vscode.Uri[]> {
+		return this._findFiles(query.pattern);
+	}
+
+	private _findFiles(query: string | undefined): vscode.Uri[] {
 		const files = this._getFiles();
 		const result: vscode.Uri[] = [];
-		const pattern = new RegExp(this._convertSimple2RegExpPattern(query.pattern));
+
+		const pattern = query ? new RegExp(this._convertSimple2RegExpPattern(query)) : null;
 
 		for (const file of files) {
-			if (pattern.exec(file.name)) {
+			if (!pattern || pattern.exec(file.name)) {
 				result.push(file.uri);
 			}
 		}
 
 		return result;
 	}
+
+	private _textDecoder = new TextDecoder();
+
+	provideTextSearchResults(query: vscode.TextSearchQuery, options: vscode.TextSearchOptions, progress: vscode.Progress<vscode.TextSearchResult>, _token: vscode.CancellationToken) {
+		const result: vscode.TextSearchComplete = { limitHit: false };
+
+		const files = this._findFiles(options.includes[0]);
+		if (files) {
+			for (const file of files) {
+				const content = this._textDecoder.decode(this.readFile(file));
+
+				const lines = content.split('\n');
+				for (let i = 0; i < lines.length; i++) {
+					const line = lines[i];
+					const index = line.indexOf(query.pattern);
+					if (index !== -1) {
+						progress.report({
+							uri: file,
+							ranges: new vscode.Range(new vscode.Position(i, index), new vscode.Position(i, index + query.pattern.length)),
+							preview: {
+								text: line,
+								matches: new vscode.Range(new vscode.Position(i, index), new vscode.Position(i, index + query.pattern.length))
```

Comment:
This is ranges within the preview, so you should just need this:

```suggestion
								matches: new vscode.Range(new vscode.Position(0, index), new vscode.Position(0, index + query.pattern.length))
```